syntax = "proto3";

package starguide;


service StarService{
    rpc Insert (CreateFavoriteUserRequest) returns (FavoriteUserResponse) {}
    rpc Delete (DeleteFavoriteUserRequest) returns (FavoriteUserResponse) {}
    rpc Find (FindFavoriteUserRequest) returns (FavoriteUsersResponse) {}
}

//Entities

message Empty {}

message FindFavoriteUserRequest{
    int32 userId = 1;
}

message CreateFavoriteUserRequest{
    int32 userId = 1;
    string productId = 2;
}

message DeleteFavoriteUserRequest{
    int32 userId = 1;
    string productId = 2;
}

message FavoriteUserResponse{
    int32 id = 1;
    int32 userId = 2;
    string productId = 3;
}

message FavoriteUsersResponse{
    repeated FavoriteUserResponse favorites = 1;
}